xtask/src/runchecks.rs

```diff
@@ -209,28 +252,41 @@ fn std_checks() {
     // for the documentation build
     env::set_var("RUSTDOCFLAGS", "-D warnings");
 
-    println!("Running std checks");
-
-    // Build each workspace
-    cargo_build(["--workspace", "--exclude=xtask"].into());
+    // Check if COVERAGE environment variable is set
+    let is_coverage = std::env::var("COVERAGE").is_ok();
 
-    // Test each workspace
-    cargo_test(["--workspace"].into());
+    println!("Running std checks");
 
     // Check format
     cargo_fmt();
 
     // Check clippy lints
     cargo_clippy();
 
+    // Build each workspace
+    cargo_build(["--workspace", "--exclude=xtask"].into());
+
     // Produce documentation for each workspace
     cargo_doc(["--workspace"].into());
 
+    // Setup code coverage
+    if is_coverage {
+        setup_coverage();
+    }
```

Comment:
The coverage got -11%, and looking at the report it seems like the reason is the lines with an `std` compilation flag are not tested (obviously), did  we setup the coverage before the `no-std` tests as well before this change?

Reply:
The problem is related to the fact that I cannot exclude `xtask` and `examples` directories anymore. So we have more files which are not tested. I cannot understand why though, perhaps `xtask` is not run on the main repository?